controllers/note.py

```diff
@@ -0,0 +1,107 @@
+import math
+
+from magicbot import StateMachine, state, timed_state, will_reset_to
+from wpimath.geometry import Translation2d
+
+from components.chassis import ChassisComponent
+from components.injector import InjectorComponent
+from components.intake import IntakeComponent
+from components.shooter import ShooterComponent
+from utilities.game import get_goal_speaker_position
+
+
+class NoteManager(StateMachine):
+
+    shooter_component: ShooterComponent
+
+    chassis: ChassisComponent
+    injector_component: InjectorComponent
+    intake: IntakeComponent
+
+    shot_desired = will_reset_to(False)
+    intake_desired = will_reset_to(False)
+    intake_cancel_desired = will_reset_to(False)
+
+    def try_intake(self):
+        self.intake_desired = True
+
+    def cancel_intake(self):
+        self.intake_cancel_desired = True
+
+    def try_shoot(self):
+        self.shot_desired = True
+
+    def has_note(self):
+        return self.injector_component.has_note()
+
+    @property
+    def translation_to_goal(self) -> Translation2d:
+        return (
+            get_goal_speaker_position().toTranslation2d()
+            - self.chassis.get_pose().translation()
+        )
```

Comment:
[PEP 8] says:

> Avoid using properties for computationally expensive operations; the attribute notation makes the caller believe that access is (relatively) cheap.

This doesn't seem _that_ cheap. I think a getter method rather than a property here makes more sense.

[PEP 8]: https://peps.python.org/pep-0008/#designing-for-inheritance